File: Assignment9/Assignment9.py
# Name: Assignment9.py
# Description: This script answers basic python syntax and operations from Assignment 9.
# All work is completed by Judah Alter.

import numpy as np
from scipy.stats import chi2

# 1a. Repeat process 100 times for Normal(0, 2^2) distribution
np.random.seed(123)
n = 20
sigma2 = 4
correct = 0

for _ in range(100):
    x = np.random.normal(0, 2, n)
    s2 = np.var(x, ddof=1)
    lower = 0
    upper = (n-1)*s2 / chi2.ppf(0.95, n-1)
    if sigma2 >= lower and sigma2 <= upper:
        correct += 1
propCorrect = correct / 100
print("Proportion correct (Normal):", propCorrect)

# 1b. Repeat process 100 times for Exponential distribution
np.random.seed(123)
correctExp = 0

for _ in range(100):
    x = np.random.exponential(scale=0.5**-1, size=n)
    s2 = np.var(x, ddof=1)
    lower = 0
    upper = (n-1)*s2 / chi2.ppf(0.95, n-1)
    if sigma2 >= lower and sigma2 <= upper:
        correctExp += 1
propCorrectExp = correctExp / 100
print("Proportion correct (Exponential):", propCorrectExp)

import numpy as np
from scipy.stats import ttest_1samp, skew
import matplotlib.pyplot as plt

# 2. Empirical power curve for t-test
np.random.seed(123)
n = 20
sd = 100
alpha = 0.05
muvals = np.arange(450, 651, 10)
power = []

for mu in muvals:
    reject = 0
    for _ in range(200):
        x = np.random.normal(mu, sd, n)
        pval = ttest_1samp(x, 500).pvalue
        if pval < alpha:
            reject += 1
    power.append(reject / 200)

plt.plot(muvals, power, marker='o')
plt.xlabel("mu")
plt.ylabel("Empirical Power")
plt.title("Empirical Power Curve")
plt.show()

# 3. Quartiles of sampling distribution of skewness statistic
np.random.seed(123)
n = 50
skewVals = []
for _ in range(500):
    x = np.random.normal(0, 1, n)
    skewVals.append(skew(x))
quartiles = np.percentile(skewVals, [25, 50, 75])
print("Quartiles of skewness statistic:", quartiles)

File: Assignment9/Assignment9.R
# Name: Assignment9.R
# Description: This script answers basic R syntax and operations from Assignment 9.
# All work is completed by Judah Alter.

# 1a. Repeat process 100 times for Normal(0, 2^2) distribution
set.seed(123)
n <- 20
sigma2 <- 4
correct <- 0

for (i in 1:100) {
  x <- rnorm(n, mean=0, sd=2)
  s2 <- var(x)
  lower <- 0
  upper <- ((n-1)*s2) / qchisq(0.95, df=n-1)
  if (sigma2 >= lower && sigma2 <= upper) {
    correct <- correct + 1
  }
}
prop_correct <- correct / 100
cat("Proportion correct (Normal):", prop_correct, "\n")

# 1b. Repeat process 100 times for Exponential distribution
set.seed(123)
lambda <- 0.5
sigma2_exp <- 4
correct_exp <- 0

for (i in 1:100) {
  x <- rexp(n, rate=lambda)
  s2 <- var(x)
  lower <- 0
  upper <- ((n-1)*s2) / qchisq(0.95, df=n-1)
  if (sigma2_exp >= lower && sigma2_exp <= upper) {
    correct_exp <- correct_exp + 1
  }
}
prop_correct_exp <- correct_exp / 100
cat("Proportion correct (Exponential):", prop_correct_exp, "\n")